src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1224,14 +1224,13 @@ void addDiagnosticForStructThis(Symbol thisParameter, int thisSlot)
 
                 if (compilation.IsFeatureEnabled(MessageID.IDS_FeatureAutoDefaultStructs))
                 {
-                    Diagnostics.Add(ErrorCode.WRN_UseDefViolationThisSupportedVersion, node.Location, thisParameter.Name);
+                    Diagnostics.Add(ErrorCode.WRN_UseDefViolationThisSupportedVersion, node.Location);
                 }
                 else
                 {
                     Diagnostics.Add(
                         ErrorCode.ERR_UseDefViolationThisUnsupportedVersion,
                         node.Location,
-                        thisParameter.Name,
```

Comment:
nit: I wonder if our diagnostic infrastructure should/could have caught that, the same way that it verifies that all parts are "printable".

Reply:
Perhaps the `DiagnosticInfo` constructor should verify `args.Length - 1` matches the largest index in the message.

Reply:
Filed #60862